### ISSUE TITLE
Hydration fails when write like "<div><span>123</span>{{value}}</div>" in the template and the data "value" is a empty string

### ISSUE THREAD

**Issue by deqwin** [vuejs/vue#11109]
### Version
2.6.11

### Reproduction link
[https://codepen.io/deqwin/pen/wvaMxoO](https://codepen.io/deqwin/pen/wvaMxoO)

### Steps to reproduce
open the reproduction above and you will see a red vue warning in the console

### What is expected?
a successful hydration

### What is actually happening?
the hydration fails and the app re-renders absolutely

---
this looks like the problem caused by the empty text node (src/core/vdom/patch.js line:646)

**Comment by enbo315** [comment]
hello  do well

**Comment by luohong123** [comment]
Hello,Modify it so that there is no error,beacuse,The client-side rendered virtual DOM tree is not matching server-rendered content. You have one more {{value}} .
```html
<div id="app" data-server-rendered="true"><span>123</span></div>
<template id="tpl">
  <div id="app"><span>123</span></div>
</template>
```

**Comment by deqwin** [comment]
yes, it works. but it's not easy to modfy because i have written this many times in my project.

**Comment by deqwin** [comment]
> Hello,Modify it so that there is no error,beacuse,The client-side rendered virtual DOM tree is not matching server-rendered content. You have one more {{value}} .
> 
> ```
> <div id="app" data-server-rendered="true"><span>123</span></div>
> <template id="tpl">
>   <div id="app"><span>123</span></div>
> </template>
> ```

**Comment by deqwin** [comment]
> Hello,Modify it so that there is no error,beacuse,The client-side rendered virtual DOM tree is not matching server-rendered content. You have one more {{value}} .
> 
> ```
> <div id="app" data-server-rendered="true"><span>123</span></div>
> <template id="tpl">
>   <div id="app"><span>123</span></div>
> </template>
> ```

Yes, it works. But it's not easy to modify because i have written this many times in my project.
